addons/casings/functions/fnc_createCasing.sqf

```diff
@@ -20,33 +20,15 @@ params ["_unit", "", "", "", "_ammo"];
 
 if (!isNull objectParent _unit) exitWith {};
 
-private _modelPath = GVAR(cachedCasings) get _ammo;
 
-if (isNil "_modelPath") then {
+private _modelPath = GVAR(cachedCasings) getOrDefaultCall [_ammo, {
     private _cartridge = getText (configFile >> "CfgAmmo" >> _ammo >> "cartridge");
-    //Default cartridge is a 5.56mm model
-    _modelPath = switch (_cartridge) do {
-        case "FxCartridge_9mm":                 { "A3\Weapons_f\ammo\cartridge_small.p3d" };
-        case "FxCartridge_65":                  { "A3\weapons_f\ammo\cartridge_65.p3d" };
-        case "FxCartridge_762":                 { "A3\weapons_f\ammo\cartridge_762.p3d" };
-        case "FxCartridge_762x39":              { "A3\weapons_f_enoch\ammo\cartridge_762x39.p3d" };
-        case "FxCartridge_93x64_Ball":          { "A3\Weapons_F_Mark\Ammo\cartridge_93x64.p3d" };
-        case "FxCartridge_338_Ball":            { "A3\Weapons_F_Mark\Ammo\cartridge_338_LM.p3d" };
-        case "FxCartridge_338_NM":              { "A3\Weapons_F_Mark\Ammo\cartridge_338_NM.p3d" };
-        case "FxCartridge_127":                 { "A3\weapons_f\ammo\cartridge_127.p3d" };
-        case "FxCartridge_127x54":              { "A3\Weapons_F_Mark\Ammo\cartridge_127x54.p3d" };
-        case "FxCartridge_slug":                { "A3\weapons_f\ammo\cartridge_slug.p3d" };
-        case "FxCartridge_12Gauge_HE_lxWS":     { "lxWS\weapons_1_f_lxws\Ammo\cartridge_he_lxws.p3d" };
-        case "FxCartridge_12Gauge_Slug_lxWS":   { "lxWS\weapons_1_f_lxws\Ammo\cartridge_slug_lxws.p3d" };
-        case "FxCartridge_12Gauge_Smoke_lxWS":  { "lxWS\weapons_1_f_lxws\Ammo\cartridge_smoke_lxws.p3d" };
-        case "FxCartridge_12Gauge_Pellet_lxWS": { "lxWS\weapons_1_f_lxws\Ammo\cartridge_pellet_lxws.p3d" };
-        case "CUP_FxCartridge_545":             { "CUP\Weapons\CUP_Weapons_Ammunition\magazines\cartridge545.p3d" };
-        case "CUP_FxCartridge_939":             { "CUP\Weapons\CUP_Weapons_Ammunition\magazines\cartridge939.p3d" };
-        case "":                                { "" };
-        default { "A3\Weapons_f\ammo\cartridge.p3d" };
+    if (_cartridge == "") then { // return (note: can't use exitWith)
+        ""
+    } else {
+        getText (configFile >> "CfgVehicles" >> _cartridge >> QGVAR(model))
```

Comment:
What about something like this?

```suggestion
        private _cartridgeConfig = configFile >> "CfgVehicles" >> _cartridge;
        private _model = getText (_cartridgeConfig >> QGVAR(model));

        if (_model == "model") then {
            _model = getText (_cartridgeConfig >> "model");

            if ("a3\weapons_f\empty" in toLowerANSI _model) then {
                _model = "";
            };
        };
        
        // Add file extension if missing (fileExists needs file extension)
        if ((_model select [count _model - 4]) != ".p3d") then {
            _model = _model + ".p3d";
        };

        ["", _model] select (fileExists _model)
```

The idea is to allow people to override the casings if necessary. `model` usually contains valid entries (the empty entry is filtered out), so instead of needing to manually configure every entry, this would allow `casing` to use already configured models.
It would avoid us (and 3rd party mods) needing to configure casings, unless specifically wanted to.

Reply:
For `FxCartridge` and `FxCartridge_Small` only, as the rest (at least in Vanilla) inherit from those. If a mod doesn't inherit from those, then it won't work.

Tbh I'm not a fan of `QGVAR(model) = "model"`, but I don't think we can check for `QGVAR(model) = ""`.

Reply:
In case people want to undefine it, I imagine they would overwrite it with `""`, intuitively.
We could force `"empty"` to signify no casing and write documentation so that people have to adopt it, but it's not very intuitive.

Reply:
fef0704dd431c9c30967d8232387a5954f0b5485 is a variation on this,
if ace_casings_model is defined then use is, otherwise just use it's normal model